packages/filesystem/src/browser/location/location-renderer.tsx

```diff
@@ -104,12 +157,88 @@ export class LocationListRenderer extends ReactRenderer {
         if (locationList) {
             const value = locationList.value;
             const uri = new URI(value);
-            this.service.location = uri;
+            this.trySetNewLocation(uri);
+            e.preventDefault();
+            e.stopPropagation();
+        }
+    }
+
+    protected onLocationKeydown(e: React.KeyboardEvent<HTMLSelectElement>): void {
```

Comment:
Since the primary effect of this method is to switch to text input from the dropdown, it might do to name it something like `toggleToTextOnKeyDown`. Might there be some way to also count the keypress? At the moment, if you hit a letter, it will transition to text input, but without the letter you pressed.

Reply:
I've updated the name of this method. I did think about the 'transition key' value but I'm not sure including the character is *really* necessary. Without a cursor, I don't know what the expected behavior *should* be. For instance, one might expect the entire field to be cleared first and then populated with the 'transition' character vs. the character being appended. Would be happy to discuss further though 😎

Reply:
That's a fair point, and I'm not really sure what the most sensible default is. I'd probably lean toward appending by default, but it might actually push me to retract the suggestion that typing should initiate a transition, since users could be disappointed in whatever default is chosen. Maybe best to just have the user trigger the transition explicitly and then decide what to do with the text input. What do you think?

Reply:
I think I'm leaning toward removing the type-to-transition functionality. It's not a very discoverable interaction since the select menu auto-expands when it receives focus, which ends up blocking the custom keyPress behavior. The added functionality might not be worth the complexity.

Reply:
I went ahead and removed this functionality